plugin-core/src/main/java/com/salesforce/bazel/eclipse/classpath/BazelClasspathContainerInitializer.java

```diff
@@ -160,11 +160,20 @@ public void requestClasspathContainerUpdate(IPath containerPath, IJavaProject ja
         }).schedule();
     }
 
-    private static void flushProjectCaches(String projectName) {
+    private static void flushProjectCaches(IProject project) {
+        // get downstream projects of the given project
+        JavaCoreHelper javaCoreHelper = BazelPluginActivator.getJavaCoreHelper();
+        IJavaProject[] allImportedProjects = javaCoreHelper.getAllBazelJavaProjects(false);
+        Set<IProject> downstreams = EclipseClasspathUtil.getDownstreamProjectsOf(project, allImportedProjects);
+
+        // flush caches
         BazelWorkspace bzlWs = BazelPluginActivator.getBazelWorkspace();
         BazelCommandManager bzlCmdMgr = BazelPluginActivator.getBazelCommandManager();
         BazelWorkspaceCommandRunner bzlWsCmdRunner = bzlCmdMgr.getWorkspaceCommandRunner(bzlWs);
-        BazelPluginActivator.getBazelProjectManager().flushCaches(projectName, bzlWsCmdRunner);
+        BazelPluginActivator.getBazelProjectManager().flushCaches(project.getName(), bzlWsCmdRunner);
+        for (IProject downstreamProject : downstreams) {
```

Comment:
If dependencies such C -> B -> A and only C and A are imported. Then A is changed. This will flush A and C. What about B?

Reply:
I think that isn't a problem, because we only load and consult aspects for C and A to setup the classpath containers for the Eclipse projects created for C and A.  Since B isn't imported, we never load aspects specifically for B (we don't have cache keys for labels of B).